lib/jbuilder.rb

```diff
@@ -270,6 +239,41 @@ def target!
 
   alias_method :method_missing, :set!
 
+  def _set(key, value = BLANK, attributes = nil)
+    result = if ::Kernel.block_given?
+      if _blank?(value)
+        # json.comments { ... }
+        # { "comments": ... }
+        _merge_block(key) { yield self }
+      else
+        # json.comments @post.comments { |comment| ... }
+        # { "comments": [ { ... }, { ... } ] }
+        _scope { _array(value) { |element| yield element } }
+      end
```

Comment:
I swapped the if condition here. It originally used to be

```ruby
      if !_blank?(value)
	# json.comments @post.comments { |comment| ... }
        # { "comments": [ { ... }, { ... } ] }
        _scope { _array(value) { |element| yield element } }
      else
        # json.comments { ... }
        # { "comments": ... }
        _merge_block(key) { yield self }
      end
```

but it is slightly faster to not negate the original check.